applications/GeoMechanicsApplication/custom_elements/transient_thermal_element.cpp

```diff
@@ -0,0 +1,642 @@
+    template<unsigned int TDim, unsigned int TNumNodes>
+    void TransientThermalElement<TDim, TNumNodes>::CalculateAndAddCapacityMatrix(
+        MatrixType& rLeftHandSideMatrix,
+        ElementVariables& rVariables)
+    {
+        KRATOS_TRY
+
+        this->CalculateCapacityMatrix(rVariables.TMatrix, rVariables);
+
+        //Distribute permeability block matrix into the elemental matrix
+        GeoElementUtilities::AssemblePBlockMatrix<0, TNumNodes>(rLeftHandSideMatrix, rVariables.TMatrix);
+
+        KRATOS_CATCH("")
+    }
+
+    // ============================================================================================
+    // ============================================================================================
+    template<unsigned int TDim, unsigned int TNumNodes>
+    void TransientThermalElement<TDim, TNumNodes>::CalculateAndAddRHS(
+        VectorType& rRightHandSideVector,
+        ElementVariables& rVariables,
+        unsigned int GPoint)
+    {
+        KRATOS_TRY
+
+        this->CalculateAndAddCapacityVector(rRightHandSideVector, rVariables);
+        this->CalculateAndAddConductivityVector(rRightHandSideVector, rVariables);
+
+        KRATOS_CATCH("")
+    }
+
+    // ============================================================================================
+    // ============================================================================================
+    template<unsigned int TDim, unsigned int TNumNodes>
+    void TransientThermalElement<TDim, TNumNodes>::CalculateKinematics(
+        ElementVariables& rVariables,
+        unsigned int PointNumber)
+    {
+        KRATOS_TRY
+
+    	//Setting the vector of shape functions and the matrix of the shape functions global gradients
+    	rVariables.N = row(rVariables.NContainer, PointNumber);
+        rVariables.GradNT = rVariables.DN_DXContainer[PointNumber];
+
+        rVariables.detJ = rVariables.detJContainer[PointNumber];
+
+        KRATOS_CATCH("")
+    }
+
+    // ============================================================================================
+    // ============================================================================================
+    template<unsigned int TDim, unsigned int TNumNodes>
+    unsigned int TransientThermalElement<TDim, TNumNodes>::GetNumberOfDOF() const
+    {
+        return TNumNodes;
+    }
+
+    // ============================================================================================
+    // ============================================================================================
+    template<unsigned int TDim, unsigned int TNumNodes>
+    void TransientThermalElement<TDim, TNumNodes>::InitializeNodalTemperatureVariables(
+        ElementVariables& rVariables)
+    {
+        KRATOS_TRY
+
+    	const GeometryType& rGeom = this->GetGeometry();
+
+        //Nodal Variables
+        for (unsigned int i = 0; i < TNumNodes; ++i) {
```

Comment:
`IndexType`

Reply:
what do you mean? We defined"
    using IndexType = std::size_t;
while TNumNodes is of type "unsigned int".
To be consistent, "unsigned int i" is applied here.